source/server/server.cc

```diff
@@ -205,7 +199,7 @@ void InstanceImpl::initialize(Options& options, TestHooks& hooks,
   // GuardDog (deadlock detection) object and thread setup before workers are
   // started and before our own run() loop runs.
   guard_dog_.reset(
-      new Server::GuardDogImpl(*admin_scope_, *config_, ProdMonotonicTimeSource::instance_));
+      new Server::GuardDogImpl(stats_store_, *config_, ProdMonotonicTimeSource::instance_));
```

Comment:
@mattklein123 The use of Stats::Scope came from your comment here: https://github.com/lyft/envoy/pull/655#discussion_r109051442

It's quite possible I misunderstood your comment then being new to Envoy but take a look back at that to make sure.

As far as the tests are concerned they verify that a mock stats scope is manipulated as expected but there isn't anything that checks which scope is passed in.